### ISSUE TITLE
Hiding/Showing array with multiple items gets out of sync with ng reactive form

### ISSUE THREAD

**Issue by obieq** [ngx-formly/ngx-formly#1450]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request
```

**Current behavior**
- I pre-load data into an array upon view load.  
- Next, I add another array item via the UI.  
- There are now two array items in both the model and the angular reactive form, which is correct.
- Separately, I have another control that, based on its value, will either hide or show the array (at this point, the array is displayed)
- I then update the other control such that it hides the form array
- I then update the other control such that it displays the form array again

Depending on where I place the form array's hideExpression, different, non-desirable behavior occurs, which I've documented in the stackblitz example.

The gist is that the model data is always correct, but the reactive form gets out of sync, which means overall form validation breaks for required fields

**Expected behavior**
Hiding and showing a form array with multiple elements should always maintain proper state for both the formly model and the angular reactive form controls

**Minimal reproduction of the problem with instructions**
https://stackblitz.com/edit/angular-w3cyfp-w1mg4v

I've outlined 3 scenarios via comments in the app.component.ts file.

Note: I've only tested w/ Chrome on macOS Mojave

**Comment by aitboudad** [comment]
some issues related to repeat field has been resolved in `V5` so here is a summary of the 3 cases:
 3 scenarios 

1. solved in  `5.0.0-rc.8`, form data is correct
2. the remove buttons should be hidden and can be solved by using `ngIf` (we should update the demo example):

```html
<div *ngIf="!field.hide" class="col-sm-2 d-flex align-items-center">
  <button class="btn btn-danger" type="button" (click)="remove(i)">Remove</button>
</div>
```

but for `Add Another Investment` buttons I'd expect that button to be visible and it should be controlled by parent field (`scenario1`) 
 
3. for `remove buttons`: it should be controlled by parent field (`scenario2`)
for  `Add Another Investment`: the same as `scenario2`

here is a working example: https://stackblitz.com/edit/angular-w3cyfp-xwkrhg

**Comment by obieq** [comment]
first and foremost, thanks for this project!

it's remarkable what we get out of the box with it (I'm evaluating it as a replacement for ng-dynamic-forms, which I had to customize heavily.  so far, no customizations required w/ our POC using this library)

also, your response times and detailed explanations to any and all submitted issues is amazing.  thank you!

I switched over to 5.0.0-rc.8 last night before I submitted the issue and still had the same behavior.  Later, I figured out how I could leverage ngIf to get the behavior I need.  However, my *ideal* scenario is #1 b/c I can just declare a hideExpression for the array and I don't have to worry about any ngIfs in my template.  

Scenario #1 does not work for me, even using your stackblitz example.  I've tested it on MacOS Safari, Chrome, and Firefox.  The data displayed in the view is correct and the model data is correct.  however, the Angular Form Control data is out of sync.

I took your stackblitz example and added some more steps outlining exactly how to reproduce the issue (and how to confirm if it's working or not).  If it works for you, then there must be some environmental issue at play.

https://stackblitz.com/edit/angular-w3cyfp-pygic6

**Comment by aitboudad** [comment]
> it's remarkable what we get out of the box with it (I'm evaluating it as a replacement for ng-dynamic-forms, which I had to customize heavily. so far, no customizations required w/ our POC using this library)

glad to hear that :+1: 

> Scenario #1 does not work for me

got it, it's a bug which should be fixed soon!

**Comment by aitboudad** [comment]
fixed in `5.0.0-rc.9`

**Note**: it safe to go with the `rc`, the stable version (`V5`) should be released within 1 or 2 weeks.
